TestControllers/AzureController.psm1

```diff
@@ -53,8 +53,22 @@ Class AzureController : TestController
 		}
 		if (!$this.OsVHD) {
 			if (($this.ARMImageName.Trim().Split(" ").Count -ne 4) -and ($this.ARMImageName -ne "")) {
-				$parameterErrors += ("Invalid value for the provided ARMImageName parameter: <'$($this.ARMImageName)'>." + `
-									 "The ARM image should be in the format: '<Publisher> <Offer> <Sku> <Version>'.")
+				if( $this.ARMImageName -Match(";") )
+				{
+					# Validate that each image is valid when heterogeneous images are used.
+					$allImageNames = $this.ARMImageName.Trim().Split(";")
+					foreach( $singleImageName in $allImageNames )
+					{
+						if(($singleImageName.Trim().Split(" ").Count -ne 4 ) -and ($singleImageName -ne "" ))
```

Comment:
it's duplicate here. You can remove line 55 and 56, and do this directly:
$allImageNames = $this.ARMImageName.Trim().Split(";") 
foreach( $singleImageName in $allImageNames ) { ... }

Reply:
Removed duplicate check. Match is unnecessary now after split on ";".